osu.Game.Rulesets.Osu/Mods/OsuModNoSliding.cs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using osu.Framework.Bindables;
+using osu.Framework.Localisation;
+using osu.Framework.Utils;
+using osu.Game.Beatmaps;
+using osu.Game.Configuration;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Osu.Beatmaps;
+using osu.Game.Rulesets.Osu.Objects;
+using osu.Game.Rulesets.Osu.Utils;
+
+namespace osu.Game.Rulesets.Osu.Mods
+{
+    public class OsuModNoSliding : Mod, IApplicableAfterBeatmapConversion
+    {
+        public override string Name => "No Sliding";
+
+        public override string Acronym => "NL";
+
+        public override double ScoreMultiplier => 1;
+
+        public override LocalisableString Description => @"Convert all sliders to streams.";
+
+        public override ModType Type => ModType.Conversion;
+
+        public override Type[] IncompatibleMods => new[] { typeof(OsuModTarget), typeof(OsuModStrictTracking) };
+
+        [SettingSource("Beat divisor")]
+        public BindableInt BeatDivisor { get; } = new BindableInt(4)
+        {
+            MinValue = 1,
+            MaxValue = 16
+        };
+
+        [SettingSource("Maximum stream length in beats", "Streams with more notes than this value will be lightened.")]
+        public BindableInt MaxStreamLength { get; } = new BindableInt(16)
```

Comment:
Not a huge fan of this setting. It's so arbitrary and needs knowledge of the map / map's timing.

Probably just remove it.